packages/design-system/src/components/Table/Table.tsx

```diff
@@ -44,7 +107,7 @@ export class Table extends React.PureComponent {
       if (
         props.scrollable &&
         Array.isArray(props.children) &&
-        !props.children.some((child) => isTableCaption(child))
+        !props.children.some((child: React.ReactElement) => isTableCaption(child))
```

Comment:
@nichia, I have noticed issues with `Table` where if I want to have an optional child, like `TableCaption`, I get errors, because assumptions are made that all children are `React.ReactElement`s. I just tried having a function that returns properly formatted markup inside a `TableCaption` optionally return `null` if there's no caption content to render, and this blows up my table. Instead, for every child, I've had to optionally return a "shell" component with `.ds-u-display--none`.

I understand that an HTML `table` can only have prescribed child nodes, in the correct order, but in React, we should be able to accept and bypass `null` for conditionally rendered children. What do you think?

Reply:
Hi @Josh68 - thanks for pick up the discrepancies. I'm not that familiar with typescript, what type definition would you suggest we use for `child` & `children`?